### PR TITLE
 Upload DefectDojo findings Action

### PR DESCRIPTION
**Pull request by carlosu7**
New action to retrieve and upload DefectDojo findings

### CODE REVIEW

File: dist/index.js
Comment (carlosu7) [inline dist/index.js:1]: This file was uploaded just for test purposes, but this doesn't need to be merged

**Review by drdavella — CHANGES_REQUESTED** [review]
@carlosu7 thanks for this work! I have confirmed that this is uploading DefectDojo results to Pixeebot once the correct URL is used. I'll be happy to approve this PR once the debugging code is cleaned up. We can add more parameters as necessary going forward.

File: src/defect-dojo.ts
Comment (drdavella) [inline src/defect-dojo.ts:60]: We can address this in a future PR.

**Comment by carlosu7** [comment]
@drdavella I have applied feedback... 

Is it OK that this PR target's [feature/defect-dojo-action](https://github.com/pixee/upload-tool-results-action/tree/feature/defect-dojo-action) or should I target to main branch?

**Comment by drdavella** [comment]
@carlosu7 let's just go ahead and rebase on `main` if that works and then I can review the entire thing.

**Comment by carlosu7** [comment]
@drdavella i updated target branch to `main`, i'm just waiting your approval

File: examples/defectdojo-pixeebot.yml
Comment (fjpgtt) [inline examples/defectdojo-pixeebot.yml:15]: Remember to update this before merging, maybe indicating main branch

**Review by fjpgtt — COMMENTED** [review]
Besides my comment everything else looks good

File: action.yml
Comment (drdavella) [inline action.yml:28]: I think this needs to be `required: false` as well,  or else it will end up being required for sonar, semgrep, etc.

**Review by drdavella — DISMISSED** [review]
Looks really good to me overall. Just one substantive comment before we can merge.

File: action.yml
Comment (carlosu7) [reply action.yml:28]: @drdavella which would be the default url?

Comment (drdavella) [reply action.yml:28]: @carlosu7 I don't think we can have a default here. The problem is that if we make this `required: true` then it will be required when the tool is `semgrep` or `sonar` which doesn't make sense.

Instead we might need to add some validation in the code like:
```python
# Please excuse my python pseudocode
if tool == "defectdojo" and not defectdojo_api_url:
    raise Error("Require API URL for tool DefectDojo")
```

Comment (carlosu7) [reply action.yml:28]: @drdavella done

**Review by fjpgtt — APPROVED** [review]

File: action.yml
Comment (JesusCotlamee) [inline action.yml:29]: I think this should be required, in https://github.com/pixee/upload-tool-results-action/pull/9/files#diff-21563dca87d9bc102a8ef9005cd9938c01e74be0aa96511fd5ee092b794f8265R45 you define throw

Comment (drdavella) [reply action.yml:29]: I am unable to find any documentation for a default URL. Unlike Sonar, I think DefectDojo is generally self-hosted so it's not like there is one canonical instance that we could refer to here.

**Review by drdavella — APPROVED** [review]

**Pull request merged**
